rules/Transform/Rector/ArrayDimFetch/ArrayDimFetchToMethodCallRector.php

```diff
@@ -63,7 +63,7 @@ public function refactor(Node $node): ?MethodCall
         }
 
         foreach ($this->arrayDimFetchToMethodCalls as $arrayDimFetchToMethodCall) {
-            if (! $this->isObjectType($node->var, new ObjectType($arrayDimFetchToMethodCall->getClass()))) {
+            if (! $this->isObjectType($node->var, $arrayDimFetchToMethodCall->getObjectType())) {
```

Comment:
@peterfox I think that `$arrayDimFetchToMethodCall->getMethod()` should be equal with `$node->dim` value, which `$node->dim->value` value need to be `String_` node, wdyt?

Reply:
Oh, I see, it used as MethodCall creation, seems ok then 👍

Reply:
@samsonasik I'm not sure I follow. Are you suggesting that the type of the ArrayDimFetch should be analysed to make sure it's a string value and if it is not a string value, to skip refactoring?

Reply:
nvm, It seems fine as it is, I just missread of how it should transform